### PR TITLE
Align ActiveReeferSettings, CargoItem, and Ref with IM

### PR DESCRIPTION
**Pull request by mp-assco**

### CODE REVIEW

File: src/main/java/org/dcsa/core/events/model/Reference.java [hunk 1]
```diff
@@ -26,6 +27,7 @@
   private UUID referenceID;
 
   @Column("reference_type_code")
+  @NotNull
   private ReferenceTypeCode referenceType;
 
   @Size(max = 100)
```
Comment (HenrikHL) [inline src/main/java/org/dcsa/core/events/model/Reference.java:33]: We need a @NotNull here

Comment (mp-assco) [reply src/main/java/org/dcsa/core/events/model/Reference.java:33]: Done

**Review by HenrikHL — DISMISSED** [review]
LGTM

File: src/main/java/org/dcsa/core/events/model/ActiveReeferSettings.java [hunk 2]
```diff
@@ -1,24 +1,36 @@
...
+import javax.validation.constraints.NotNull;
 import javax.validation.constraints.Size;
 import java.util.UUID;
 
 @NoArgsConstructor
 @Data
-public class ActiveReeferSettings extends AuditBase {
+@Table("active_reefer_settings")
+@EqualsAndHashCode(callSuper = true)
+public class ActiveReeferSettings extends AuditBase implements Persistable<UUID> {
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/core/events/model/ActiveReeferSettings.java:23]: Could you remove AuditBase?  we do not use it.

File: src/main/java/org/dcsa/core/events/model/ActiveReeferSettings.java [hunk 3]
```diff
@@ -1,24 +1,36 @@
...
 
     @Id
     /* We do not show this in JSON as it is an internal detail */
     @Column("shipment_equipment_id")
     private UUID shipmentEquipmentID;
 
+    @Transient
+    @JsonIgnore
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/core/events/model/ActiveReeferSettings.java:31]: No need for @JsonIgnore,  we have separated entities and DTOs

**Review by douradofrazer — CHANGES_REQUESTED** [review]

File: src/main/java/org/dcsa/core/events/model/ActiveReeferSettings.java (hunk 3 above)
Comment (mp-assco) [reply src/main/java/org/dcsa/core/events/model/ActiveReeferSettings.java]: Done

File: src/main/java/org/dcsa/core/events/model/ActiveReeferSettings.java (hunk 2 above)
Comment (mp-assco) [reply src/main/java/org/dcsa/core/events/model/ActiveReeferSettings.java]: Done

**Review by pimg — APPROVED** [review]

**Pull request merged**